Produlytics/ui/back-end/src/main/java/it/deltax/produlytics/uibackend/repositories/RilevazioneRepository.java

```diff
@@ -12,14 +14,15 @@
 @Repository
 public interface RilevazioneRepository extends CrudRepository<DetectionEntity, DetectionEntityId> {
 
-    // autoimplementazione da parte di spring
-    // *Id*DeviceId perché fa parte della chiave composta (DetectionEntityId)
-    List<DetectionEntity> findByIdDeviceIdAndIdCharacteristicId(int deviceId, int characteristicId, Sort sort);
-
+    @Query("""
+        select d from DetectionEntity d
+        where d.id.deviceId = :deviceId
+            and d.id.characteristicId = :characteristicId
+            and (creation is null or d.id.creationTime > :creation)""")
```

Comment:
@SkiFire13 

Qui sta il check in jpa che dicevo (questa non è una query nativa, viene interpretata da jpa)

Reply:
Ok, ma rimane il problema di aggiornare le rilevazioni vecchie solo quando vengono richieste quelle più nuove di una certa data.